### PR TITLE
119/r4/typography update

### PR DESCRIPTION
**Pull request by akdetrick**
## Description
Rebuilt typography system to reflect new design standards.

- New dictionary vars and values for type size and color; updated media sizing
- Updated utility classes to reflect the new system
- Added typography "spec sheet" to documentation

### New type vars in dictionary
Screen Shot 2019-08-08 at 2 15 42 PM
Screen Shot 2019-08-08 at 2 16 00 PM

### Spec sheet
Screen Shot 2019-08-08 at 2 08 12 PM
Screen Shot 2019-08-08 at 2 06 45 PM
Screen Shot 2019-08-08 at 2 07 04 PM
Screen Shot 2019-08-08 at 2 07 17 PM
Screen Shot 2019-08-08 at 2 07 30 PM
Screen Shot 2019-08-08 at 2 08 04 PM
Screen Shot 2019-08-08 at 2 08 20 PM

### Updated media sizing

**New**
Screen Shot 2019-08-08 at 3 33 36 PM

**Old**
Screen Shot 2019-08-08 at 5 31 49 PM

### Updated utility classes
Screen Shot 2019-08-08 at 3 34 01 PM
Screen Shot 2019-08-08 at 3 33 51 PM
Screen Shot 2019-08-08 at 3 34 12 PM

## Checklist
- [x] Docs have been rebuilt (`yarn build:full`)
- [x] All unit tests pass
- [x] Snapshots have been updated
- [x] No lint errors or warnings have been introduced
- [x] **[Version bumped](https://github.com/cbinsights/form-design-system#updating-version-number) if appropriate**

### CODE REVIEW

File: src/dictionary/properties/font/size.json [hunk 1]
```diff
@@ -2,13 +2,15 @@
...
-      "tiny":     { "value": "10px" }
+      "xs":      { "value": "10px" },
+      "s":       { "value": "12px" },
+      "m":       { "value": "14px" },
+      "default": { "value": "{font.size.m.value}" },
+      "l":       { "value": "16px" },
+      "xl":      { "value": "20px" },
+      "2xl":     { "value": "24px" },
+      "3xl":     { "value": "32px" },
+      "4xl":     { "value": "38px" }
```
Comment (rey-wright) [inline src/dictionary/properties/font/size.json]: 👍

File: src/components/style/interactive/toast.css [hunk 2]
```diff
@@ -21,7 +21,7 @@
 
 .toast-action {
   color: var(--font-color-light) !important;
-  font-size: var(--font-size-small) !important;
+  font-size: var(--font-size-s) !important;
```
Comment (rey-wright) [inline src/components/style/interactive/toast.css]: 👍 I see the t shirt standardization

File: src/dictionary/properties/font/color.json [hunk 3]
```diff
@@ -1,12 +1,24 @@
...
 
+      "heading":    { "value": "{color.base.charcoal.value}" },
+      "primary":    { "value": "rgba(64, 64, 64, 0.87)" },
+      "secondary":  { "value": "rgba(64, 64, 64, 0.60)" },
+      "hint":       { "value": "rgba(64, 64, 64, 0.38)" },
+
+      "headingInverted":    { "value": "{color.base.white.value}" },
+      "primaryInverted":    { "value": "rgba(255, 255, 255, 0.87)" },
+      "secondaryInverted":  { "value": "rgba(255, 255, 255, 0.60)" },
+      "hintInverted":       { "value": "rgba(255, 255, 255, 0.38)" },
```
Comment (rey-wright) [inline src/dictionary/properties/font/color.json]: nice, I see we're using the same alpha transparency scales as the non inverted ones too.

**Comment by akdetrick** [comment]
Replying here for posterity.

Regarding design/product vision, there are relatively few places where we will use inverted context. There are a few reasons why I included support for inversion here:

- The `primary`/`secondary`/`hint` concepts are more scalable than the existing "dark, default, light" concepts. We have room to add in a `tertiary` if we want.
- Design has already expressed an interest in alpha-based text colors; if we're doing it for dark text on light backgrounds, it also makes sense to introduce support for light text on dark backgrounds
- While we don't use inverted context much right now in product, having the standards recorded in the design system opens up the design team to more creative thinking around how we use it
- Smaller components in the design system can be composed with `inverted` in mind (e.g. adding a `Chip` theme for filter chips, which will likely be dark with light text)

**Pull request merged**